dcpy/utils/versions.py

```diff
@@ -89,6 +89,38 @@ def bump(
         return Quarter(year=new_year, quarter=new_quarter)
 
 
+@dataclass(order=True)
+class Date(Version):
+    year: int
+    month: int
+    day: int
+
+    @property
+    def label(self) -> str:
+        return f"{self.year}-{self.month:02d}-{self.day:02d}"
+
+    def bump(
+        self, version_subtype: VersionSubType | None = None, bump_by: int | None = None
+    ) -> Date:
+        if version_subtype is not None:
+            raise Exception(
+                f"Version subtype {version_subtype} not applicable for Date versions"
+            )
+        bump_by = bump_by or 1
+        current_date = datetime.strptime(self.label, "%Y-%m-%d")
+        end_date = current_date + timedelta(days=bump_by)
+        return Date(year=end_date.year, month=end_date.month, day=end_date.day)
+
+    @staticmethod
+    def generate() -> Date:
+        version = parse(date.today().strftime("%Y-%m-%d"))
```

Comment:
should we just call this something like `from_todays_date()` and just use this as a static constructor? (as opposed to calling the version parser below)

Reply:
FYI, looks like `FirstOfMonth` has that same `generate` circuitous parsing logic. Would be nice to apply the same convention there. (not a blocker)